Fio dashboard changes

andreyvEze opened the pull request.

trukhilio commented on src/fio/fioEngine.js:
```diff
@@ -118,173 +113,9 @@ export class FioEngine extends CurrencyEngine {
     this.recentFioFee = { publicAddress: '', fee: 0 }
 
-    this.fioSdkInit()
+    // this.fioSdkInit()
```
Should be removed or not?

trukhilio commented on src/fio/fioEngine.js:
```diff
@@ -347,7 +178,7 @@ export class FioEngine extends CurrencyEngine {
     }
 
-    await this.checkAbiAccounts()
+    // await this.checkAbiAccounts()
```
Should be removed or not?

peachbits committed:
Upgrade react-native-zcash

peachbits committed:
Remove extra 'started' engine variable in favor of 'engineOn'

peachbits committed:
Upgrade react-native-piratechain

peachbits committed:
Remove old checkpoint files

peachbits committed:
Use scan progress reported directly from synchronizer

Downloading and scanning happen simultaneously and the progress is reported in scanProgress. The percentages can remain the same

peachbits committed:
Upgrade react-native-zcash

The new sdk reports balances changes and transactions as they are found so we don't have to query for them ourselves. The newWallet boolean is hardcoded for now and will be updated in the next commit for clarity

peachbits committed:
Use scanProgress directly for balance and transaction status

peachbits committed:
Update initializer config params

The sdk wants to know if this is a new wallet or an existing one. Also, we're changing the alias so it's truly unique

peachbits committed:
Don't store our public key as receive address

The pubkey used used to be the static shielded address but were storing the actual view key now. And since the sdk doesn't reveal the recipient (unless we sent the tx) we don't know which address the funds were received at

peachbits committed:
Update default fee and add `fee` and `raw` from SDK to `processTx`

Jon-edge committed:
Support 12/24-word mnemonics for XLM

peachbits committed:
Return Birthday Height with private seed

peachbits committed:
Allow 1% progress updates but no more often than 1 second

swansontec committed:
v2.5.5

Jon-edge committed:
Add EdgeTxAction to TronEngine

swansontec committed:
Add a CLI for working with plugins

peachbits committed:
Merge pull request #635 from EdgeApp/matthew/upgradePirate

Update and re-enable Piratechain

peachbits committed:
v2.6.0

peachbits committed:
Sanity check initial progress report

We can't trust the first scanProgress emitted from the sdks. Sometimes it's abnormally large and then drops down, presumably as it's figuring out just how far behind it is from the network height. We can throw this away. If it returns 100 though we can keep it

peachbits committed:
Update Piratechain lightwalletd server url

peachbits committed:
Remove unused types

peachbits committed:
Use types exported from react-native-zcash

peachbits committed:
Add autoshielding

We need to overstuff syncNetwork In order to get access to the keys to create an autoshielding transaction. To initiate one, we identify in the balance event that we have excess transparent funds and then force syncNetwork to restart and then create the transaction. Once we see it mined we can release the lock and continue looking for excess transparent funds

Commit:
File sync from EdgeApp/edge-workflows

peachbits committed:
Merge pull request #639 from EdgeApp/matthew/zcash2.0

Upgrade react-native-zcash

Jon-edge committed:
XRP  EdgeTxAction Partial Fill

peachbits committed:
Add hack for missing memos from shielded transactions on Android

The Android SDK isn't returning memos for shielding transactions and there isn't another way to identify these transactions. This hack will track these messed up transactions locally (at least until a resync or reinstall) so they appear

Jon-edge committed:
Upgrade edge-core-js to v1.10.0

Jon-edge committed:
Merge pull request #638 from EdgeApp/jon/trx-txinfo

- added: EdgeTxAction tagging to TRX freeze/unfreeze contract call transactions

Jon-edge committed:
Merge pull request #637 from EdgeApp/jon/xlm-mnemonic

- added: Support for importing XLM wallets via 12/24-word mnemonic seed phrase

peachbits committed:
Merge pull request #642 from EdgeApp/matthew/zcashAutoshield

Add autoshield

peachbits committed:
v2.7.0

peachbits committed:
Update Zcash address explorer url

peachbits committed:
Allow fast 100% updates through

The isSynced and addressesChecked conditional isn't necessary with the progress ratio checks

peachbits committed:
Upgrade react-native-zcash to v0.6.2

samholmes committed:
Add Calibration Filecoin Testnet

peachbits committed:
Fix lifecycle

We need to be able to call makeSynchronizer after calling killEngine

samholmes committed:
Add additional query during scanning to account for incoming transfers

samholmes committed:
Reverse Filfox scanning order

samholmes committed:
Always call `onScan` between scanning queries

peachbits committed:
Merge pull request #643 from EdgeApp/matthew/zcashFollowup

Zcash v2 followup fixes

peachbits committed:
v2.7.1

samholmes committed:
Make sure to kill Filfox scanning when engine is killed

samholmes committed:
Progress lastAddressQueryHeight during Filfox scanning

samholmes committed:
Use transfers API from Filfox for Filecoin tx history

samholmes committed:
Add deprecation comment for scanTransactionsFromFilscan

peachbits committed:
Fix memos

peachbits committed:
Merge pull request #647 from EdgeApp/matthew/zcashMemo

Fix Zcash memos

samholmes committed:
Fix Filecoin getMaxSpendable

Continuously query networkFees using `makeSpend` until `tx.networkFee`
is stable before returning the `spendableBalance` amount.

samholmes committed:
Add balance check to Filecoin makeSpend

paullinator committed:
Fix Filecoin `makeSpend` fee estimation

samholmes committed:
Merge pull request #646 from EdgeApp/sam/fil-fee-estimation

Fix Filecoin `makeSpend` fee estimation

samholmes committed:
Merge pull request #648 from EdgeApp/sam/fil-max-spend

Fix Filecoin getMaxSpendable

peachbits committed:
Use destructuring

swansontec committed:
Provide proper type exports for this library

peachbits committed:
Attempt to fix 0 date race condition

If the transaction listener hears about the tx before the sending function resolves we should replace the 0 date with the current time so it appears at the top of the transaction list

samholmes committed:
Add networkPrefix to FilecoinNetworkInfo

samholmes committed:
Rename `FilfoxMessageDetailed` type identifiers

samholmes committed:
Avoid over-processing messages in `scanTransactionsFromFilfox`

peachbits committed:
Init Cosmos Tools and Engine

peachbits committed:
Add key derivation utils

peachbits committed:
Add parseUri and encodeUri

peachbits committed:
Add shared tool client and queryBalance

peachbits committed:
Add Thorchain

peachbits committed:
Merge pull request #650 from EdgeApp/matthew/zcashDateSpecialHandling

Fix new zcash tx 0-date race condition

peachbits committed:
v2.7.2

samholmes committed:
Merge pull request #645 from EdgeApp/sam/filecoin-testnet

Add Calibration Filecoin Testnet

samholmes committed:
Merge pull request #644 from EdgeApp/sam/filecoin-tx-history

Improve Filecoin transaction querying

samholmes committed:
v2.8.0

swansontec committed:
Merge pull request #651 from EdgeApp/william/fix-packaging

Provide proper type exports for this library

swansontec committed:
v2.8.1

swansontec committed:
Delete empty GitHub workflow

samholmes committed:
Use a hacky, but reliable, getMaxSpendable technique for Filecoin

samholmes committed:
Add missing set of `engineOn` within `FilecoinEngine.startEngine`

samholmes committed:
Wait until scanning is aborted before resolving `killEngine`

peachbits committed:
Create script to generate Thorchain typescript files

samholmes committed:
Merge pull request #653 from EdgeApp/sam/fil-resync-bugs

Filecoin resync bugs

samholmes committed:
Merge pull request #652 from EdgeApp/sam/filecoin-max-spend

Fix getMaxSpendable for Filecoin

samholmes committed:
v2.8.2

Jon-edge committed:
Merge pull request #641 from EdgeApp/jon/xrp-txinfo

Implement XRP DEX order parsing -> EdgeTxAction's

Jon-edge committed:
v2.9.0

peachbits committed:
Add custom type handling

Each Cosmos-based plugin needs to augment the Registry so the SDK can handle its unique protobuf definitions. CosmosMethods exposes a common api (so far, just a single 'transfer' method) that each plugin can then use to create the appropriate encoded object.

peachbits committed:
Add makeSpend, signTx, and broadcastTx

peachbits committed:
Add queryTransactions

There are rpc methods that can retrieve transactions but testing showed that not all nodes can retrieve historical data. More discovery is necessary to identify archive nodes.

peachbits committed:
Add txid caching to exit early from queryTransactions

peachbits committed:
Add blockheight query

peachbits committed:
Use walletId for synchronizer alias

On disk data issues can cause syncing loops that are only resolvable by reinstall. This change will allow import key to be an alternative troubleshooting step but at the cost of forcing each wallet to redownload and sync to initialize a synchronizer with a new alias

samholmes committed:
Fix contract address for PYUSD

Jon-edge committed:
Tx Action Swap Direction Partial Fix
Fix for only the bugged src/dest direction fix.

Partial fill amounts that happen after the initial order tx will require
a future fix and api change.

Jon-edge committed:
Merge pull request #657 from EdgeApp/jon/fix/xrp-txaction

XRP EdgeTxActionSwap Direction Fix

Jon-edge committed:
v2.9.1

peachbits committed:
Merge pull request #655 from EdgeApp/matthew/pirateAlias

Use walletId for synchronizer alias

samholmes committed:
Upgrade @ethereumjs dependencies

samholmes committed:
Support type 2 EVM transaction with EIP1559 parameters

peachbits committed:
v2.9.2

Jon-edge committed:
Use eth-balance-checker contracts for batch token balance queries

peachbits committed:
Convert ARRR alias to hex

samholmes committed:
Added support to send to 0x addresses for Filecoin plugin

samholmes committed:
Remove duplicate identifiers in EthereumNetwork

samholmes committed:
Fix `explicit-function-return-type` lint errors in EthereumNetwork

samholmes committed:
Fix `QueryFuncs` type in EthereumNetwork

samholmes committed:
Fix `strict-boolean-expressions` lint errors in EthereumNetwork

peachbits committed:
Merge pull request #660 from EdgeApp/matthew/fixArrrAlias

Convert ARRR alias to hex

peachbits committed:
v2.9.3

samholmes committed:
Fix invalid typescript syntax

samholmes committed:
Fix `typescript-eslint/default-param-last` in EthereumNetwork

samholmes committed:
Merge pull request #656 from EdgeApp/sam/pyusd

Fix contract address for PYUSD

samholmes committed:
Merge pull request #658 from EdgeApp/sam/1559-tx-type

Support type 2 EVM transaction with EIP1559 parameters

samholmes committed:
v2.9.4

peachbits committed:
Update bridged USDC currency code to USDC.e

peachbits committed:
Test evmscan balance responses for integer string

swansontec committed:
Update Travis build environment

swansontec committed:
Merge pull request #649 from EdgeApp/william/cli

Add a CLI for working with plugins

Jon-edge committed:
Merge pull request #659 from EdgeApp/jon/batch-balance-query

On-Chain Batch Balance Query

Jon-edge committed:
v2.10.0

peachbits committed:
Initialize this.progressRatio in constructor and after resync

This decouples the progress from the synchronizer and better represents the state of the engine

peachbits committed:
Only update progress if number go up

Also allow more incremental 1% updates

peachbits committed:
Restart the synchronizer if it stops responding

A bug in the synchronizer block downloader logic will cause all activity to stop completely without warning or errors. It just stops downloading and parsing blocks and hangs. Since the synchronizer is noisy at boot we can restart it if we haven't heard from it in while.

It's unknown if the bug occurs after the initial sync is complete and the synchronizer is just listening for the latest blocks. I'm assuming it does not and am limiting this method to just initial sync.

peachbits committed:
Merge pull request #654 from EdgeApp/matthew/cosmos

Add Thorchain (RUNE)

peachbits committed:
Merge pull request #663 from EdgeApp/matthew/theFonz

Restart synchronizer if it stop providing status updates

peachbits committed:
Merge pull request #661 from EdgeApp/matthew/usdce

Update bridged USDC currency code to USDC.e

peachbits committed:
v2.11.0

peachbits committed:
Add MsgDeposit support to Thorchain registry

peachbits committed:
Move MakeTx types to common

peachbits committed:
Fix path

peachbits committed:
Merge pull request #664 from EdgeApp/matthew/updateImportPath

Fix CosmosEngine import path

samholmes committed:
Use transaction type 0 for currencies which do not support EIP-1559

peachbits committed:
Break out unsigned tx creation

peachbits committed:
Add MakeTxDeposit type

samholmes committed:
Merge pull request #665 from EdgeApp/sam/eip1559-fixes

Use transaction type 0 for currencies which do not support EIP-1559

samholmes committed:
v2.11.1

samholmes committed:
Removed unsafeBroadcastTx configuration for Filecoin

samholmes committed:
Create a separate method for checking multiple token balances

samholmes committed:
Refactor `QueryFuncs` into `NetworkAdapter` type

samholmes committed:
Factor out overloaded multicastServers impl. to respective methods

samholmes committed:
Separate Ethscan and RPC blockheight methods

samholmes committed:
Add type parameter to `promiseAny`

samholmes committed:
Factor out overloaded multicastServers impl. for broadcastTx

samholmes committed:
Factor out network specific methods to `NetworkAdapter` interfaces

samholmes committed:
Rename methods on NetworkAdapter

This makes the method names clearer that they're are retrieval methods.

samholmes committed:
Rename `multicastServers` to `multicastRpc` to reflect it's new purpose

samholmes committed:
Abstract multi-server requests to two protected utility methods

paullinator committed:
Fix failed sync of evm wallets

samholmes committed:
Change `multicastRpc` argument signature to match RPC call signature

samholmes committed:
Collapse multicastRpc method implementation cases

Now that the call signature for `EthereumNetwork.multicastRpc` matches
`NetworkAdaptor.multicastRpc`, there is no need for the case handling
for each method.

samholmes committed:
Filecoin FEVM currencies

samholmes committed:
Consolidate server config to a new `NetworkAdapterConfig` type

samholmes committed:
Merge pull request #668 from EdgeApp/paul/fixEvmSync

Fix failed sync of evm wallets

paullinator committed:
v2.11.2

peachbits committed:
Misc cleanups

peachbits committed:
Replace shapeshift query with native RPC node transaction query

peachbits committed:
Add apikey to client initialization

samholmes committed:
Merge pull request #662 from EdgeApp/sam/filecoin-fevm

Add Filecoin FEVM

samholmes committed:
v2.12.0

peachbits committed:
Use fetchCors in Cosmos SDK

peachbits committed:
Remove hardcoded mainnet denom

peachbits committed:
Add Osmosis

Osmosis MsgSend matches the default Cosmos type so no entry in the registry is necessary. We can break out the default action and leave the switch/case for non-standard types for each plugin

samholmes committed:
Fix currencyCode symbol for filecoinfevmcalibration

samholmes committed:
Fix token balance query in RpcAdapter

samholmes committed:
Add evmMemoOptions to Filecoin FEVM currencies

This enables smart contract data to be sent to a Filecoin FEVM wallet

samholmes committed:
Add iFIL and WFIL tokens to filecoinfevm(calibration)

samholmes committed:
Fix Filecoin f4 address parsing

samholmes committed:
Properly handle missing `fetchTxs` adapter method

Instead of using the EvmScanAdapter's fetchTxs as a hack around, we
check for `fetchTxs` functionality across all our adapters and handle
the case where this is missing. We must handle the case as an empty
transaction result set to allow the engine to fully sync.

samholmes committed:
Simplify NetworkAdapter interface by making it an abstract class

The `NetworkAdapterBase` becomes an abstract class, `NetworkAdapter`,
and extended classes can implement this abstract class with all the
utilities included with the class along with the documentation on the
abstract fields required to be implemented.

This is a lot nicer/easier to work with, within a good IDE like VSCode.
It also makes it simpler because now we can avoid the
`extends...implements` with just a single `extends`.

peachbits committed:
Merge pull request #669 from EdgeApp/matthew/cosmosFetch

Add API keys to client initialization

peachbits committed:
Merge pull request #667 from EdgeApp/matthew/MsgDeposit

Add Thorchain MsgDeposit support

peachbits committed:
Update progress after each query

peachbits committed:
Group clients together

We'll expose another client when estimating fees

peachbits committed:
Don't export networkInfo

Some fantom network fee test uses it so we can still export when testing

peachbits committed:
Don't export currencyInfo

Unless we need it for testing

Jon-edge committed:
Add multicastEthProviders utility function

Jon-edge committed:
Add resolveEnsName method to EthereumTools otherMethods
This method is specific to ENS handling on the Ethereum network

paullinator committed:
Ensure at least a 2 GWEI 'tip' to miners

This check prevents the maxPriorityFeePerGas from going negative which would error in creating the transaction.

samholmes committed:
Merge pull request #672 from EdgeApp/paul/minFeeTip

Ensure at least a 2 GWEI 'tip' to miners

peachbits committed:
Add chain registry data

Instead of duplicating this data in network info we can just use the registry directly. We'll keep a default set local and attempt to update it when the tools are initialized

Jon-edge committed:
Merge pull request #666 from EdgeApp/jon/ens-name-res

ENS Name Resolution

Jon-edge committed:
v2.13.0

peachbits committed:
Add CosmosTxOtherParams types

createUnsignedTxHexOtherParams shouldn't have return an object it just makes it a little harder to add other stuff to it. Since it's changing the name should, too

peachbits committed:
Break out fee calculation

An anemic helper for now, this will expand as new chains and assets are added

peachbits committed:
Add gas limit and gas price to fee calculation

paullinator committed:
Merge pull request #671 from EdgeApp/matthew/nativeTxQuery

Replace Shapeshift API with native RPC searchTx

peachbits committed:
Add fee cache

The 'simulate' method is a network call. Since the gasLimit is an overestimation (and gas prices are static) we can cache the returned object

peachbits committed:
Split transaction query into archive and validator queries

Add support for an additional archive rpc and only look back when necessary. We're assuming a validator will have 2 weeks of transaction data and only use the archive node during the initial sync and then again only if 2 weeks have passed

peachbits committed:
Fix deposit alias

paullinator committed:
Merge pull request #674 from EdgeApp/matthew/fixDepositAlias

Fix deposit alias

paullinator committed:
v2.13.1

paullinator committed:
Add use of Pokt RPCs for fantom and polygon

paullinator committed:
Update Avalanche to use avascan.info

paullinator committed:
Merge pull request #675 from EdgeApp/paul/morePokt

Add use of Pokt RPCs for fantom and polygon

paullinator committed:
Merge pull request #676 from EdgeApp/paul/avaxEvmScan

Update Avalanche to use avascan.info

paullinator committed:
v2.14.0

peachbits committed:
Resolve eslint error suppression

peachbits committed:
Resolve typescript error suppression

peachbits committed:
Add spl token balance query

Add bulk rpc fetch ability to make the token balance fetching more efficient

peachbits committed:
Add spl token spending

peachbits committed:
Convert queryTransactions to bulk query

peachbits committed:
Add token tx processing

The intermediate function finds all the token balance changes in a transaction and then calls processTx for each unique token

peachbits committed:
Add token

peachbits committed:
Refresh rpc node list

Also make sure to throw if a successful 200 response returns an error

Jon-edge committed:
Support new core 'auto-tokens'

peachbits committed:
Replace token balance query

The previous one uses a "secondary index" which is expensive and not all nodes support it. This one is more direct but requires we derive the underlying address first

peachbits committed:
Make coin sent and receive events parsing more robust

This looks at all the events in the transaction and passes each relevant balance change to processTx.

Previous implementation only looked at first transfer event but that is only sufficient for simple sends. Using coin_spent and coin_received will see additional transaction types where coins are moved

peachbits committed:
Add token balance query and transaction loop progress updating

peachbits committed:
Add common makeSpend functionality to CurrencyEngine

peachbits committed:
Add token spending

paullinator committed:
Update goerli

Add pokt rpc server and tokens

samholmes committed:
Merge pull request #670 from EdgeApp/sam/filecoin-fevm-fixes

Filecoin FEVM fixes

samholmes committed:
v2.15.0

peachbits committed:
Merge pull request #673 from EdgeApp/matthew/osmosis

Add Osmosis

peachbits committed:
Add Osmosis token ION

peachbits committed:
Add Coreum

peachbits committed:
Fix switched thorchain node urls

peachbits committed:
Add unconfirmed transaction handling

Cosmos-based chains usually have fast confirmations but since transaction use a nonce we need to track pending txs. The engine will use a calculated nonce value that considers how many transactions have been seen (minus ones we processed) and the nonce that was present when we sent the first unconfirmed. expired unconfirmed transactions are removed occasionally based on a (totally arbitrary) 30 minute timeout.

paullinator committed:
Add missing RUNE symbol

peachbits committed:
Merge pull request #681 from EdgeApp/matthew/coreum

Add Coreum

peachbits committed:
Merge pull request #683 from EdgeApp/matthew/cosmosFixes

(cosmos-based) Add unconfirmed nonce handling

trukhilio committed:
Remove unused code

paullinator committed:
Merge pull request #680 from EdgeApp/paul/updateGoerli

Update goerli

Jon-edge committed:
Upgrade edge-core-js@^1.12.0

paullinator committed:
Implement Cosmos getMaxTx

Jon-edge committed:
Merge pull request #678 from EdgeApp/jon/core-detected-tokens

Jon/core-detected-tokens

Jon-edge committed:
v2.16.0

paullinator committed:
Merge pull request #684 from EdgeApp/paul/runeSymbol

Add missing RUNE symbol

peachbits committed:
Split up tx query into smaller chunks

Requesting 1000 transactions is kind of a lot. Splitting it up means a better chance of success and provides more meaningful progress updates. 50 is arbitrary.

paullinator committed:
Merge pull request #685 from EdgeApp/paul/cosmosGetMaxTx

Implement Cosmos getMaxTx

trukhilio committed:
Remove ethereum plugin. Update tests.

trukhilio committed:
Remove unused scripts and files.

trukhilio committed:
Remove unused packages. Update and lock other packages.

peachbits committed:
Disable Pokt for 'getTransaction' method, for now

It's returning all the correct txids but some of them can't be queried with getTransaction.

peachbits committed:
Apply apikey to archive node url

peachbits committed:
Merge pull request #679 from EdgeApp/matthew/solanaTokens

Solana: Add token support

peachbits committed:
Merge pull request #682 from EdgeApp/matthew/cosmosTokens

Add token support (Cosmos-based)

peachbits committed:
Fix missing makeMetaTokens

peachbits committed:
Merge pull request #686 from EdgeApp/matthew/thorSync

Apply apikey to archive node url

peachbits committed:
v2.17.0

peachbits committed:
Fix Solana token currency code

peachbits committed:
Fix typo in RPC method

samholmes committed:
Fix CHANGELOG entry for 2.12

peachbits committed:
Merge pull request #688 from EdgeApp/matthew/fixTypoInRpc

Fix 'latests' typo

trukhilio committed:
Don't run syncNetwork and checkBlockchainInnerLoop, checkAccountInnerLoop on prod because we don't need it.

trukhilio committed:
Merge pull request #6 from trukhilio/fio-dashboard

FIO plugin update

trukhilio committed:
Update to 2.17 version

trukhilio committed:
Remove android and ios files

trukhilio committed:
Merge pull request #7 from trukhilio/fio-dashboard

Update to 2.17

trukhilio committed:
Remove extra calls to FIO servers on FioTools constructor.

trukhilio committed:
Merge pull request #8 from trukhilio/fio-dashboard

Remove extra calls to FIO servers on FioTools constructor.

trukhilio committed:
DASH-1270 Remove unused code

trukhilio committed:
Update @greymass/eosio patch from 0.6.8 to 0.6.11 version.

trukhilio committed:
Add @types/minimatch:5.1.2 version.

trukhilio committed:
Add types:[] to tsconfig because of global packages issue.
